### ISSUE TITLE
permission to fork OSIRIS DRP pytest setup for use with GPI DRP?

### ISSUE THREAD

**Issue by mperrin** [Keck-DataReductionPipelines/OsirisDRP#39]
Hello OSIRIS team, 

The GPI team would like to borrow much of your unit test automation (from https://github.com/Keck-DataReductionPipelines/OsirisDRP/tree/develop/tests) for use with the GPI DRP. Since we have the same exact setup with XML files etc, much of what you've written can be used by us with minimal adaptation. 

I checked for a software license on the OSIRIS pipeline but can't find one, so I want to check this reuse is OK with you all. We will of course give credit in the GPI pipeline docs and changelog. The GPI DRP is released under a BSD 3-clause license; see https://github.com/geminiplanetimager/gpi_pipeline/blob/master/LICENSE.txt 
We will be most grateful if you confirm that we can reuse your tests code under that license.

Thanks! @mperrin and @kfollette for the GPI DRP team

**Comment by followthesheep** [comment]
This sounds fine to me. Anyone else have an opinion about this? @jlyke-keck @alexrudy @jluastro?

I think you bring up a good point about the license. I don't believe we ever discuss this. Maybe we can discuss this at the next OSIRIS pipeline meeting in two weeks.

**Comment by jluastro** [comment]
We should definitely add a license; but I am fairly agnostic about the choice of license (MIT?). But we need to get feedback from James L. and Keck who will know who the “owners” of the code are and what license the code was developed under.

Cheers,
Jessica

**Comment by alexrudy** [comment]
I'm definitely 👍 on other projects re-using this. I'm license agnostic too.

Also, @mperrin, I modified https://github.com/Keck-DataReductionPipelines/OsirisDRP/blob/develop/backbone/drpBackbone__define.pro to work with the testing framework, so you'll have to fold in similar modifications as well, which let the drpBackbone exit after consuming a queue, instead of running forever in test mode.

**Comment by mperrin** [comment]
Thanks all.   My suspicion is the original code probably doesn't have a well-defined license - there's a ton of IDL code where people were too casual about releasing code without explicitly licensing it. Myself included if we go back far enough... 

@alexrudy FYI we added years ago to the GPI pipeline a feature to run one recipe specified on the command line and then immediately exit. Along with lots of other substantial refactoring and reorganization of the backbone. The GPI pipeline is a descendent of the OSIRIS one but definitely long since diverged from having a very close resemblance under the hood...

Meanwhile @kfollette and I got a first set of tests working on Wednesday at our team meeting. We ended up borrowing less code than I had expected, mostly some pytest setup and fixtures. For all the queue handling and recipe code we already had our own Python equivalents to those so we didn't need to borrow any of that.
